### PR TITLE
ocrvs-5562 form validations

### PR DESCRIPTION
**Pull request by Zangetsu101**
Farajaland PR: https://github.com/opencrvs/opencrvs-farajaland/pull/667

### CODE REVIEW

File: packages/config/src/handlers/forms/validation.ts [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+                .flatMap((group) => group.fields)
+                .map(({ mapping }) => mapping?.template?.fieldName)
+            ])
+            .filter((maybeName): maybeName is string => Boolean(maybeName))
+        )
+        return {
+          message: `All the certificate handlebars should be unique for an event. Duplicate handlebars for "${duplicateCertificateHandlebars.join(
+            ', '
+          )}"`
+        }
```
Comment (euanmillar) [inline packages/config/src/handlers/forms/validation.ts:414]: @Zangetsu101 I fixed all the certificate handlebars in [this PR](https://github.com/opencrvs/opencrvs-core/pull/5785). But this validation still throws an error even if the duplicateCertificateHandlebars array is empty

**Review by euanmillar — CHANGES_REQUESTED** [review]

File: packages/config/src/handlers/forms/validation.ts [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+              ...(sec.mapping?.template ?? []).map(
+                ({ fieldName }) => fieldName
+              ),
+              ...sec.groups
+                .flatMap((group) => group.fields)
+                .map(({ mapping }) => mapping?.template?.fieldName)
+            ])
+            .filter((maybeName): maybeName is string => Boolean(maybeName))
+        ).length === 0
+      },
```
Comment (Zangetsu101) [inline packages/config/src/handlers/forms/validation.ts:396]:
```suggestion
      (sections) =>
        findDuplicates(
          sections
            .flatMap((sec) => [
              ...(sec.mapping?.template ?? []).map(
                ({ fieldName }) => fieldName
              ),
              ...sec.groups
                .flatMap((group) => group.fields)
                .map(({ mapping }) => mapping?.template?.fieldName)
            ])
            .filter((maybeName): maybeName is string => Boolean(maybeName))
        ).length === 0,
```
@euanmillar Sorry my bad. I wasn't returning the predicate

Comment (euanmillar) [reply packages/config/src/handlers/forms/validation.ts:396]: Works perfectly @Zangetsu101 .  I applied the fix in my PR.

File: packages/config/src/handlers/forms/validation.ts [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+        )
+        .refine(
+          (sec) => {
+            if (!REQUIRED_FIELDS_IN_SECTION[sec.id]) {
+              return true
+            }
+            const nonCustomfieldsInSection = sec.groups.flatMap((group) =>
+              group.fields
+                .filter(({ custom }) => !Boolean(custom))
+                .filter(({ type }) => type !== 'DIVIDER')
```
Comment (Zangetsu101) [inline packages/config/src/handlers/forms/validation.ts:330]: I think we need to filter all the `viewOnly` fields here

Comment (euanmillar) [reply packages/config/src/handlers/forms/validation.ts:335]: OK.  Thats probably a minor hotfix if we get around to it

**Review by euanmillar — APPROVED** [review]

**Pull request merged**